File: hamcode 6/275770.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "regexp"
)

func main() {
    reader := bufio.NewReader(os.Stdin)
    s, _ := reader.ReadString('\n')
    
    p := regexp.MustCompile(`(kalan(\s*)tar)`)
    var resultSt [][2]int
    
    for {
        matches := p.FindAllStringSubmatchIndex(s, -1)
        if len(matches) == 0 {
            break
        }
        
        var t, c int
        var newS string
        lastIdx := 0
        
        for _, match := range matches {
            start := match[0]
            end := match[1]
            spaceStart := match[4]
            spaceEnd := match[5]
            
            newS += s[lastIdx:start]
            
            if spaceEnd-spaceStart == 0 {
                t++
            } else {
                c++
            }
            
            lastIdx = end
        }
        
        newS += s[lastIdx:]
        s = newS
        resultSt = append(resultSt, [2]int{t, c})
    }
    
    fmt.Println(len(resultSt))
    for _, pair := range resultSt {
        fmt.Println(pair[0], pair[1])
    }
}

File: hamcode 6/275770.py
import re

s = input()

p = re.compile(r"(kalan(\s*)tar)")
result_st = []
while True:
    ma = list(p.finditer(s))
    if not ma:
        break
    t = 0
    c = 0
    prt = []
    last = 0
    for m in ma:
        start, end = m.span()
        prt.append(s[last:start])
        spaces = m.group(2)
        if len(spaces) == 0:
            t += 1
        else:
            c += 1
        last = end
    prt.append(s[last:])
    s = "".join(prt)
    result_st.append((t, c))

print(len(result_st))
for x, y in result_st:
    print(x, y)
